### ISSUE TITLE
Fix S2589 FP: comparison inside a `while (true)` loop.

### ISSUE THREAD

**Issue by fiotti** [SonarSource/sonar-dotnet#8401]
### Description

False positive S2589: comparison inside a `while (true)` loop.

### Repro steps

```csharp
class SonarTest {
    void Something() {
        // example
    }

    public void Test() {
        int iterations = 5;
        while (true) {
            iterations--;

            Something();

            if (iterations == 0) // S2589: Change this condition so that it does not always evaluate to 'False'.
                break;
        }
    }
}
```

### Expected behavior

I didn't expect any diagnostics to be emitted.

### Actual behavior

False positive.

### Known workarounds

```csharp
#pragma warning disable S2589
```

### Related information

* SonarLint for Visual Studio 2022, version 7.4.0.80741
* Visual Studio Professional 2022 (64 bit), version 17.7.6
* dotnet 7.0.403
* MSBuild version 17.7.2+d6990bcfa for .NET Framework, version 17.7.2.37605
* Windows 11 Pro 22H2 22621.2715

**Comment by Tim-Pohlmann** [comment]
Hi @fiotti and thanks for the report. Loops with counters give the analyzer some trouble. We are looking into improving the behavior. The issue is tracked here: #8028
